js/widgets/modewidget.js

```diff
@@ -9,152 +9,145 @@
 // License along with this library; if not, write to the Free Software
 // Foundation, 51 Franklin Street, Suite 500 Boston, MA 02110-1335 USA
 
-function ModeWidget() {
-    const ROTATESPEED = 125;
-    const BUTTONDIVWIDTH = 535;
-    const BUTTONSIZE = 53;
-    const ICONSIZE = 32;
+class ModeWidget {
+
+    get CONSTANTS() {
+        return {
+            ICONSIZE: 32,
+            BUTTONSIZE: 53,
+            ROTATESPEED: 125,
+            BUTTONDIVWIDTH: 535
+        }
+    }
```

Comment:
`static` would've been cleaner, innit?

Performance difference here would be negligible but I personally think that's a better practice.

Reply:
By `static` you meant to have static class fields for make the getter `static`? I think `static` class fields can be changed and are not constant. I am inclined towards having multiple 1 line getters, 1 for each constant.

It would look something like this
```
static ICONSIZE(){ return 32; }
static BUTTONSIZE(){ return 53; }
static ROTATESPEED(){ return 125; }
static BUTTONDIVWIDTH(){ return 535; }
```
Then we could cleanly access the constants using the classname dot the constant. What do you think?

Reply:
```
class FooBar {
    static ICONSIZE = 32;
    ...
    foo() {
        ...
        ... FooBar.ICONSIZE ...
        ...
    }
    ...
}
```

Reply:
> ```
> static ICONSIZE(){ return 32; }
> static BUTTONSIZE(){ return 53; }
> static ROTATESPEED(){ return 125; }
> static BUTTONDIVWIDTH(){ return 535; }
> ```

this is overkill in my opinion.